src/Shared/DiagnosticSourceInstrumentation/PropertyFetcher.cs

```diff
@@ -26,6 +27,9 @@ namespace OpenTelemetry.Instrumentation;
 /// <typeparam name="T">The type of the property being fetched.</typeparam>
 internal sealed class PropertyFetcher<T>
 {
+#if NET6_0_OR_GREATER
+    private const string TrimCompatibilityMessage = "PropertyFetcher access property on the object type; therefore is not trim compatible.";
```

Comment:
nit: Feels kind of awkward. Here is my attempt at making it better:

`PropertyFetcher is used to access properties on objects dynamically by design and cannot be made trim compatible`